### ISSUE TITLE
Is Connection._client truly thread-safe?

### ISSUE THREAD

**Issue by dustymugs** [pynamodb/PynamoDB#914]
I'm in the process of seeing how thread-safe pynamodb is and a question came up regarding `Connection._client`...

https://github.com/pynamodb/PynamoDB/blob/master/pynamodb/connection/base.py#L522

Let's think of a scenario with Thread 1 and Thread 2 with the same 

1. Thread 1 works upon ModelA with a variety of actions (e.g. get, query, save)
2. At some point after Thread 1 is done... Thread 2 starts working on ModelA

What would be the state of `self._client` for Thread 2 as the client was constructed when Thread 1 was acting upon ModelA?

**Comment by ikonst** [comment]
Are you talking specifically about multi-threading rather than userspace threads? We use PynamoDB regularly with gevent without any issues, but I'm not familiar thread-safety in boto -- would need to learn more.

**Comment by dustymugs** [comment]
No, I'm not talking specifically about multi-threading vs userspace threads as I use both (threading.Thread and async/await). The big question mark I see is that given `self._client` is assigned an object from one thread's `botocore` session, what does that do for another thread using that same client object?

**Comment by dustymugs** [comment]
One reason for my concern is that my code uses userspace threads to load many instances of a polymorphic model with discriminators.
